ldap/servers/plugins/acl/acl.c

```diff
@@ -176,6 +179,60 @@ check_rdn_access(Slapi_PBlock *pb, Slapi_Entry *e, const char *dn, int access)
     return (retCode);
 }
 
+/* Retries in the targetfilter cache (list) if this
+ * filter has already been evaluated
+ */
+static struct targetfilter_cached_result *
+targetfilter_cache_lookup(struct acl_pblock *aclpb, char *filter)
+{
+    struct targetfilter_cached_result *results;
+    if (!config_get_targetfilter_cache()) {
+        /* targetfilter cache is disabled */
+        return NULL;
+    }
+    if (filter == NULL) {
+        return NULL;
+    }
+    for(results = aclpb->aclpb_curr_entry_targetfilters; results; results = results->next) {
+        if (strcasecmp(results->filter, filter) == 0) {
+            return results;
+        }
+    }
+    
+    return NULL;
+}
+
+/* Free all evaluations cached for this current entry */
+void
+targetfilter_cache_free(struct targetfilter_cached_result **per_entry_results)
+{
+    struct targetfilter_cached_result *results, *next;
+    for(results = *per_entry_results; results;) {
+        next = results->next;
+        slapi_ch_free((void **) &results->filter);
```

Comment:
Minor nit pick, but to be compliant with our coding guidelines the char "filter" should be freed using "slapi_ch_free_string()"

Reply:
@mreynolds389 Thanks for you review. I will fix that. In addition I think the config parameter 'nsslapd-targetfilter-cache' should be set once (for example when the aclpb is allocated) else even if it is atomic read it can create fake contention on it during parallel evaluation of targetfilters.